### PR TITLE
Revert "Disable faulty Windows Clang job"

### LINKED ISSUE #2664: Revert #2661
## Subject of the issue

https://github.com/SFML/SFML/pull/2661 is a workaround and needs to be reverted at some point once the Windows images are updated to use Clang 16. Hopefully this happens in a few more days but may take a few more weeks.

### PR DESCRIPTION
**Pull request by ChrisThrasher**
## Description

Closes #2664 

This reverts commit bbb6f60dda95a1c63372ab60061f60cf3fa4f9c7.

They're finally shipping Clang 16 in these images so we can restore these jobs.

### CODE REVIEW

**Review by eXpl0it3r — APPROVED** [review]
Feel free to merge if your confident on the stability 😄

**Comment by ChrisThrasher** [comment]
Multiple projects of mine are using these updating runners with Clang 16 but I may still hold off one more day before merging to be safe. There's reason we must merge this today.

**Pull request merged**
